### PR TITLE
increase font size accross admin, improve color contrast

### LINKED ISSUE #5270: Fix font size and contrast of text across all of the admin
### Issue Summary

Follow-up of #2224. Related: #5269. There are still a lot of places in the admin UI where text does not have enough contrast compared to its background, either because the text is too small (font size and weight), or the color does not have a good contrast ratio against the background color.

There is no WCAG 2.1 requirement for a specific font size, the requirement is for a [minimum contrast ratio](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html).

Suggested approach:

- [x] Identify parts of the UI where fonts are too small / do not use the right color to provide enough contrast
- [x] Suggest increasing the size, weight, or change color
- [x] Potentially define a minimum font size to use across the whole UI

---

Here are the parts of the UI I could think of as being too small:

* [ ] Table headings
* [ ] Page status ("Live" / "Draft")
* [ ] “Modified” timestamp in page edit footer

For the contrast, here are parts of the UI that have been flagged in automated testing as too low contrast:

- [ ] "Release notes" link in the "Upgrade notification" block on the dashboard (admin only), path: /
- [ ] "Success" banner messages in the admin, white text and buttons on green background, e.g. after "Publish" on a page
- [ ] "Configure a site now" CTA at root level of the listing pages, path: /pages/
- [ ] "Warning" banner messages in the admin, white text and buttons on yellow background, e.g. when reviewing a revision /pages/60/revisions/37/revert/
- [ ] "Replace current draft" CTA of review revisions, path: /pages/60/revisions/37/revert/
- [ ] Pages "move" UI when some pages cannot receive sub-pages, 
- [ ] Page "parent" link in search results view of the link chooser, on hover
- [ ] "Max dimensions" and "File size" labels in image editing UI

There might be more that I missed. What the automated tools don't find is cases where the color contrast is good if you just compare text color with that of its background, but the text is too small or thin.

Annotated screenshots:

Dashboard – Dashboard

User account – Current time zone - Success

Pages – View child pages - Root level

Pages – Review revision

review revisions cta

Pages – Move

Link chooser search page parent hover

Images – Edit

### PR DESCRIPTION
**Pull request by hippogriffic**
Fixes #5270.

- add new color-grey variable, adjust existing variables
- increase body font size to 85%
- add 20px to sidebar width
- adjust layout on image focal point label to avoid text overlap
- add new dark color variables and use for improved contrast on text

browser testing in progress now, but @thibaudcolas if you could take a look in the meantime I would appreciate it

### CODE REVIEW

File: client/scss/settings/_variables.scss [hunk 1]
```diff
@@ -86,7 +91,7 @@ $font-serif: Roboto Slab, Georgia, serif;
 
 // misc sizing
 $thumbnail-width: 130px;
-$menu-width: 180px;
+$menu-width: 200px;
```
[the remark this answers is not in the document]
Comment (thibaudcolas) [reply client/scss/settings/_variables.scss]: (I think I might indeed be too pessimistic, this seems to be working fine as far as I can tell).

**Comment by thibaudcolas** [comment]
Spotted one instance that still has the same issue as from #5270:

chooser-modal-contrast

File: client/scss/settings/_variables.scss [hunk 2]
```diff
@@ -29,13 +29,17 @@ $breakpoints: (
...
 $color-teal-darker: darken(adjust-hue($color-teal, 1), 4);
 $color-teal-dark: darken(adjust-hue($color-teal, 1), 7);
 
 $color-blue: #71b2d4;
 $color-red: #cd3238;
+$color-red-dark: #b4191f;
 $color-orange: #e9b04d;
+$color-orange-dark: #b67f19;
 $color-green: #189370;
+$color-green-dark: #157b57;
```
[the remark this answers is not in the document]
Comment (hippogriffic) [reply client/scss/settings/_variables.scss]: that was my first thought, however there are instances where these colour variables are used with `lighten` or `darken` and changing the value doesn't work well, for example:
image

changing the original red makes that highlighted background very dark. Adding new variables seemed like the simpler solution than digging through the code to find all these instances

File: client/scss/settings/_variables.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (hippogriffic) [reply client/scss/settings/_variables.scss]: yeah, with the overall body test increase the sidebar was a little tight, I was concerned that this change would have larger knock on effects but it seems to cope quite well. There was a slight issue with the layout on the image editor but that was causing problems on some screen sizes anyway and there was a pretty simple fix.

File: client/scss/settings/_variables.scss [hunk 3]
```diff
@@ -29,13 +29,17 @@ $breakpoints: (
...
 $color-teal: #007d7e;
+$color-teal-accessible: #006465;
 $color-teal-darker: darken(adjust-hue($color-teal, 1), 4);
 $color-teal-dark: darken(adjust-hue($color-teal, 1), 7);
 
 $color-blue: #71b2d4;
 $color-red: #cd3238;
+$color-red-dark: #b4191f;
 $color-orange: #e9b04d;
+$color-orange-dark: #b67f19;
```
[the remark this answers is not in the document]
Comment (hippogriffic) [reply client/scss/settings/_variables.scss]: this was the color Katie recommended but I'm happy to darken it a bit more

File: client/scss/settings/_variables.scss [hunk 4]
```diff
@@ -29,13 +29,17 @@ $breakpoints: (
 
 // colours
 $color-teal: #007d7e;
+$color-teal-accessible: #006465;
 $color-teal-darker: darken(adjust-hue($color-teal, 1), 4);
 $color-teal-dark: darken(adjust-hue($color-teal, 1), 7);
```
[the remark this answers is not in the document]
Comment (hippogriffic) [reply client/scss/settings/_variables.scss]: this is a good point, `color-teal-darker` outputs as `#00676a` and has a pretty good contrast ratio, I'll remove the new variable and use this one for link text

**Comment by hippogriffic** [comment]
I've pushed some changes, including a fix for the above issue

File: client/scss/elements/_typography.scss [hunk 5]
```diff
@@ -84,7 +84,7 @@ dl {
 }
 
 dt {
-    color: darken($color-grey-3, 15%);
+    color: $color-grey-3;
```
Comment (thibaudcolas) [inline client/scss/elements/_typography.scss]: From what I can see this new shade of grey is only used here. This really doesn't feel worth the extra color – would be fine to just go with `$color-grey-2`.

**Comment by thibaudcolas** [comment]
This is working very well, I was initially a bit concerned about the global font size increase but it actually feels great to use.

**Comment by thibaudcolas** [comment]
Merged! This will be released in Wagtail 2.6.

Here is a before/after GIF for posterity:

color-contrast-ftw

**Pull request merged**